app/controllers/submissions_controller.rb

```diff
@@ -495,8 +495,8 @@ def get_file
         end
 
         if SubmissionFile.is_binary?(file_contents)
-          # If the file appears to be binary, send it as a download
-          render plain: 'not a plaintext file'
+          # If the file appears to be binary, display a warning
+          render json: { content: I18n.t('submission_file.error.binary_file_message').to_json, type: 'error' }
```

Comment:
Sorry, I was wrong about this. type should be `'unknown'`, otherwise the syntax highlighter won't be able to handle it.